src/main/config.cpp

```diff
@@ -0,0 +1,84 @@
+#include <iostream>
+#include <sstream>
+#include "gflags/gflags.h"
+#include "config.hpp"
+
+extern "C" {
+	#include "exec_monitor.h"
+}
+
+/*
+	FLAGS SECTION
+*/
+
+// EXEC FLAG - MONITOR PROCESS EXECUTION
+DEFINE_bool(exec, false, "Monitors process executions.");
+DEFINE_string(ppid, "", "Specifies the ppid(s) of the processes to monitor using comma-separated values.");
+DEFINE_string(name, "", "Specifies the name(s) of the processes to monitor using comma-separated values.");
+DEFINE_int32(n, 0, "Specifies the number of processes to monitor. The program will stop after n processes were executed.");
+DEFINE_int32(t, 0, "Specifies the number of seconds to monitor. The program will stop after t seconds.");
+
+// DATA EXPORT FLAGS
+DEFINE_string(save, "", "Specifies the format in which to save the terminal output in a separate file. Available formats are csv and protobuf.");
+DEFINE_validator(save, Config::checkSaveAndExportFormat);
+DEFINE_string(export, "", "Specifies the format in which to export data in a separate file. Using export will not print to stdout. Available formats are csv and protobuf.");
+DEFINE_validator(export, Config::checkSaveAndExportFormat);
+
+// END OF FLAGS SECTION
+
+Config::Config()
+{
+}
+
+void Config::detect_usecase(int argc, char *argv[])
+{
+	gflags::SetUsageMessage("");
```

Comment:
A usage message could be helpful to figure out how to use hawk? (ie: it's not immediately clear that there is one mandatory flag, either --exec or --kernel_module etc... and then less important flags (such as -n etc)